### PR TITLE
CAMEL-11017 - move empty directories between consumers and producers

### PR DESCRIPTION
**Pull request by onders86**
UT's provided for;

file component 
ftp component
sftp component

### CODE REVIEW

File: components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/SftpOperations.java [hunk 1]
```diff
@@ -714,97 +722,104 @@ private boolean retrieveFileToStreamInBody(String name, Exchange exchange) throw
...
-            local = new File(local, relativeName);
-
-            // create directory to local work file
-            local.mkdirs();
-
-            // delete any existing files
-            if (temp.exists()) {
-                if (!FileUtil.deleteFile(temp)) {
-                    throw new GenericFileOperationFailedException("Cannot delete existing local work file: " + temp);
+        if (file.getFile() != null && file.getFile() != null
```
Comment (duchuyctlk) [inline components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/SftpOperations.java]: Is `file.getFile() != null` duplicated intently? And `file.getFile().getAttrs()` is not null-checked before you invoke a method call on it.

File: components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/SftpOperations.java [hunk 2]
```diff
@@ -714,97 +722,104 @@ private boolean retrieveFileToStreamInBody(String name, Exchange exchange) throw
...
+                }
+            
+                channel.get(remoteName, os);
+            
+            } catch (SftpException e) {
+                LOG.trace("Error occurred during retrieving file: {} to local directory. Deleting local work file: {}", name, temp);
+                // failed to retrieve the file so we need to close streams and delete in progress file
+                // must close stream before deleting file
+                IOHelper.close(os, "retrieve: " + name, LOG);
+                boolean deleted = FileUtil.deleteFile(temp);
```
Comment (duchuyctlk) [inline components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/SftpOperations.java]: the boolean `deleted` is in use only once inside the conditional statement below so I think it could be omitted like line 759 and line 817:
```java
if (!FileUtil.deleteFile(temp)) {
```

File: components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/SftpOperations.java (hunk 1 above)
Comment (onders86) [inline components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/SftpOperations.java]: well spotted. thanks

File: components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/SftpOperations.java (hunk 2 above)
Comment (onders86) [inline components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/SftpOperations.java]: i think you miss something reviewing. seems irrelevant comment

**Comment by onders86** [comment]
Hi,

I want to add one more option to zip & tar data format for split EIP based on the feature in this PR. If you can review and looks good for you, I want to work on that as well. The feature in this PR works only if URI option specified. i believe it won't break anything. @davsclaus could you have a look? (I am aware you would not like to see update in core component like this. Please review and let me know if it's something which may not be required.)

Thanks.

**Comment by onders86** [comment]
@davsclaus any update?

**Comment by onders86** [comment]
@davsclaus , thanks for the feedback at JIRA. Closing
